packages/@sanity/form-builder/src/inputs/ObjectInput/ObjectInput.tsx

```diff
@@ -80,6 +84,56 @@ export const ObjectInput = memo(
       filterField = DEFAULT_FILTER_FIELD,
     } = props
 
+    const inputId = useId()
+    const filterGroups: FieldGroup[] = React.useMemo(() => {
+      if (!type.groups || type.groups.length === 0) {
+        return []
+      }
+
+      return [
+        {
+          name: DEFAULT_FIELD_GROUP_NAME,
+          title: 'All fields',
+          fields: type.fields,
+        },
+        ...(type.groups || []),
+      ]
+    }, [type.groups, type.fields])
+    const defaultFieldGroupName = React.useMemo(() => {
+      if (filterGroups.length === 0) {
+        return DEFAULT_FIELD_GROUP_NAME
+      }
+
+      return (
+        defaultTo(
+          find(filterGroups, (fieldGroup) => fieldGroup.default),
+          filterGroups?.[0]
+        )?.name || DEFAULT_FIELD_GROUP_NAME
+      )
+    }, [filterGroups])
+    const [selectedFieldGroupName, setSelectedFieldGroupName] = useState(defaultFieldGroupName)
+    const [isAllGroupsHidden, setIsAllGroupsHidden] = useState(false)
+    const fieldGroupRootFocusPaths = React.useMemo(() => {
+      if (filterGroups.length === 0) {
+        return type.fields.map((field) => field.name)
+      }
+
+      return (
+        find(filterGroups, (fieldGroup) => fieldGroup.name === selectedFieldGroupName)?.fields || []
```

Comment:
any reason to use `lodash.find` here? I think it's sensible to prefer native array methods unless there's a clear benefit with using lodash.